bears/vcs/mercurial/HgCommitBear.py

```diff
@@ -0,0 +1,210 @@
+import re
+import shutil
+import os
+
+from urllib.parse import urlparse
+from coala_utils.ContextManagers import change_directory
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Result import Result
+from bears.vcs.CommitBear import _CommitBear
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+class HgCommitBear(_CommitBear):
+    LANGUAGES = {'Hg'}
+    SUPPORTED_HOST_KEYWORD_REGEX = {
+        'bitbucket': (r'[Cc]los(?:e[sd]?|ing)'
+                      r'|[Rr]esolv(?:e[sd]?|ing)'
+                      r'|[Ff]ix(?:e[sd]|ing)?')
+    }
+
+    @classmethod
+    def get_shortlog_checks_metadata(cls):
+        return FunctionMetadata.from_function(
+            cls.check_shortlog,
+            omit={'self', 'shortlog'})
+
+    @classmethod
+    def get_body_checks_metadata(cls):
+        return FunctionMetadata.from_function(
+            cls.check_body,
+            omit={'self', 'body'})
+
+    @classmethod
+    def get_issue_checks_metadata(cls):
+        return FunctionMetadata.from_function(
+            cls.check_issue_reference,
+            omit={'self', 'body'})
+
+    @classmethod
+    def get_metadata(cls):
+        return FunctionMetadata.merge(
+            FunctionMetadata.from_function(
+                cls.run,
+                omit={'self', 'dependency_results'}),
+            cls.get_shortlog_checks_metadata(),
+            cls.get_body_checks_metadata(),
+            cls.get_issue_checks_metadata())
+
+    @classmethod
+    def check_prerequisites(cls):
+        if shutil.which('hg') is None:
+            return 'hg is not installed.'
+        else:
+            return True
+
+    @staticmethod
+    def get_host_from_remotes():
+        """
+        Retrieve the first host from the list of mercurial remotes.
+        """
+        remotes, _ = run_shell_command('hg paths')
+
+        remotes = [url.split()[-1] for url in remotes.splitlines()]
+        if len(remotes) == 0:
+            return None
+
+        url = remotes[0]
+        if 'hg@' in url:
+            netloc = re.findall(r'@(\S+)\.', url)[0]
+        else:
+            netloc = urlparse(url)[1]
+        return netloc.split('.')[0].split('@')[-1]
+
+    def run(self, allow_empty_commit_message: bool = False, **kwargs):
+        """
+        Check the current mercurial commit message at HEAD.
+
+        This bear ensures automatically that the shortlog and body do not
+        exceed a given line-length and that a newline lies between them.
+
+        :param allow_empty_commit_message: Whether empty commit messages are
+                                           allowed or not.
+        """
+        with change_directory(self.get_config_dir() or os.getcwd()):
+            stdout, stderr = run_shell_command(
```

Comment:
if you can move this to a new method, the `run` could be moved to the base class...?

Reply:
Ah
Yeah sure
I will correct this
Thanks a ton ! :D